### PR TITLE
CV2-4386 work on allowing async

### PR DESCRIPTION
**Pull request by DGaffney**
## Description
Small tweaks to allow async on image and audio but keep legacy for text and video

Reference: CV2-4386

## How has this been tested?
Tested locally and appears to be working fine.

## Have you considered secure coding practices when writing this code?
None come to mind here

### CODE REVIEW

File: app/main/lib/image_similarity.py
Comment (DGaffney) [reply app/main/lib/image_similarity.py:67]: Not refactoring this at this time - feel free to push back on this though!

**Review by skyemeedan — COMMENTED** [review]
This looks fine, and I can approve this to unblock if needed.  I don't feel like I personally have a solid enough understanding of how the sync to async transition works (we've discussed in some calls, but I don't think it is written down on tickets?) to say this ready for production.  Specifically, if lots of async requests start failing, I'm assuming we will be losing data.  This an appropriate/acceptable trade off for occasional failures, but do we have monitoring in place to know if it is happening at a large scale? (like if service down?)

File: app/main/controller/similarity_sync_controller.py
Comment (DGaffney) [reply app/main/controller/similarity_sync_controller.py:33]: will fix!

File: app/main/controller/similarity_sync_controller.py
Comment (DGaffney) [reply app/main/controller/similarity_sync_controller.py:31]: These are leftovers I need to just cull

File: app/main/lib/image_similarity.py
Comment (DGaffney) [reply app/main/lib/image_similarity.py:91]: We've historically applied 0.0 throughout alegre - I think it's safe to continue applying that in this case but @computermacgyver should weigh in!

File: migrations/versions/774d0d537e69_update_key_length_on_image.py
Comment (DGaffney) [reply migrations/versions/774d0d537e69_update_key_length_on_image.py:1]: This shouldn't cause any issues with previous keys - at this level of the system, these are simply strings, and we are expanding the max length of the possible strings. This did not cause any issues before because 64 was plenty of space for `["check", "project_media", project_media.id, field_name]`, base-64 encoded, which is what all strings were historically. In this update, we needed to add the idea of a "temporary" ID which is just a SHA256 of the URL (see the line `project_media.id = Digest::MD5.hexdigest(project_media.url).to_i(16)` in Check API / `alegre_v2.rb`). The size of that ID is, what, 33 orders of magnitude larger or so than our current IDs, so its just such a large (fake and temporary but technically an integer that will also never collide) number that it forces us to expand the string space. These are short-lived objects, but while alive we need to accommodate their ridiculous strings. There are probably better ways of re-architecting lots of this, but we're already on a side-quest from the alrger goal and I wanted to figure out a straightforward way to keep us as on track as ossible.

File: app/main/lib/image_similarity.py
Comment (computermacgyver) [reply app/main/lib/image_similarity.py:91]: `0` is all images in the context. In practice we're always including a threshold when Check API uses the code. I think `0` is a good default if a threshold is not given. My own concern would be whether we error out because the response is too large or something. I think, however, we're fine at present

**Review by computermacgyver — APPROVED** [review]

**Comment by computermacgyver** [comment]
Great work here. I think we should go ahead and merge this to develop (provided the related PR for Check API is set and approved). We have good logging in Check API at present. I think we will want to improve logging and retries here as well in the future, but I don't think this PR fundamentally changes what happens when a service is down/non-responsive.

**Comment by DGaffney** [comment]
> Great work here. I think we should go ahead and merge this to develop (provided the related PR for Check API is set and approved). We have good logging in Check API at present. I think we will want to improve logging and retries here as well in the future, but I don't think this PR fundamentally changes what happens when a service is down/non-responsive.

Sounds good! We will merge once the latest check-api changes are finished up!

**Pull request merged**
